WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -285,5 +285,59 @@ public static async Task<ISet<uint256>> GetAllDependentsAsync(this IRPCClient rp
 
 			return hashSet;
 		}
+
+		public static async Task<VerboseBlockInfo> GetVerboseBlockAsync(this IRPCClient rpc, uint256 blockId)
+		{
+			var resp = await rpc.SendCommandAsync(RPCOperations.getblock, blockId, 3).ConfigureAwait(false);
+			var blockInfoStr = resp.Result.ToString();
+			var blockInfoJson = JObject.Parse(blockInfoStr);
+			var previousblockhash = blockInfoJson.Value<string>("previousblockhash");
+			var blockInfo = new VerboseBlockInfo()
+			{
+				Hash = uint256.Parse(blockInfoJson.Value<string>("hash")),
+				PrevBlockHash = previousblockhash is { } ? uint256.Parse(previousblockhash) : null,
+				Confirmations = blockInfoJson.Value<ulong>("confirmations"),
+				Height = blockInfoJson.Value<ulong>("height"),
+				BlockTime = Utils.UnixTimeToDateTime(blockInfoJson.Value<uint>("time"))
+			};
+
+			foreach (var txJson in blockInfoJson["tx"])
+			{
+				var tx = new VerboseTransactionInfo()
+				{
+					Id = uint256.Parse(txJson.Value<string>("txid"))
+				};
+
+				foreach (var txinJson in txJson["vin"].Where(x => x["coinbase"] is null))
+				{
+					var input = new VerboseInputInfo()
+					{
+						OutPoint = new OutPoint(uint256.Parse(txinJson.Value<string>("txid")), txinJson.Value<uint>("vout")),
+						PrevOutput = new VerboseOutputInfo()
```

Comment:
I think we should consider to not deserialize PrevOutputs if theirs ScriptType (it is in the json) is not a segwit script because we are decoding many millons of scripts/values and allocating memory as crazy.

Reply:
What about an optional param in the method signature `segwitonly` will that work?

Reply:
I am okay with that but I don't think we will use this anywhere else anyway, that's why I didn't deserialized all the fields that come in the json and focused only on those that are useful for your work.